src/python/pants/backend/python/target_types.py

```diff
@@ -673,4 +673,7 @@ async def inject_dependencies(
 
 
 def rules():
-    return collect_rules()
+    return (
+        *collect_rules(),
+        UnionRule(InjectDependenciesRequest, InjectPythonDistributionDependencies),
+    )
```

Comment:
Oops...we should probably cherry-pick this portion of the change to 2.0.

Reply:
When we went with the `@union` and `UnionRule` split, it was because `collect_rules()` wasn't a thing. I wonder if there is an improved factoring (`@union` and `@union_member` ?) that would still be compatible with both collection and explicit listing...